### PR TITLE
Document archive-node redundancy

### PR DESCRIPTION
**Pull request by deepthiskumar**
Usage docs for archive-node redundancy

### CODE REVIEW

File: docs/archive-redundancy.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+2. For extensional blocks: (Generated from option 3)
+
+    To import extensional blocks generated from an archive database, specify the `--extensional` flag in the `archive_blocks` command
+
+## Staking ledgers
+
+Staking ledgers are used to determine slot winners for each epoch. Mina daemon stores staking ledger for the current epoch and the next epoch (after it is finalized). Whenever there is an epoch transition, the "next" staking ledger from the previous epoch is used to determine slot winners for the new epoch. The staking ledger from the previous epoch is replaced with the "next" staking ledger and a new "next" staking ledger is chosen. Since staking ledgers for older epochs are no longer accessible, users may want to still keep them around for reporting or other purposes.
+
+Currently these ledgers can be exported using the cli command:
+`coda.exe ledger export CURRENT-STAGED-LEDGER|STAKING-EPOCH-LEDGER|NEXT-EPOCH-LEDGER`
```
Comment (psteckler) [inline docs/archive-redundancy.md:61]: the names of the ledgers need to be all-lowercase

I think it's a bug in the `help` system that prints them in uppercase

File: docs/archive-redundancy.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    If you provide both hashes, you get all blocks between start and end, and including the start and end. If just the end hash is provided, then it generates blocks starting with the unparented block closest to the end block, possibly the genesis block. The tool generates a file with name `<protocol-state-hash>.json` for each block. The block data in these files are called extensional blocks. Since these are generated from the database, they would have only the data stored in the archive database and would not contain any other information pertaining to a block (for example, blockchain snark)
+
+    Currently this tool is not packaged and can only be build from source using `make missing_subchain` 
+
+## Identifying missing blocks
+
+The tool `missing_block_auditor.exe` can be used to determine any missing blocks in an archive database. The tool outputs a list of state hashes of all the blocks that are missing. This can be used to monitor the archive database for any missing blocks.
```
Comment (psteckler) [inline docs/archive-redundancy.md:40]: To be precise, the tool identifies missing parent blocks. There can be other missing blocks in subchains that end in the missing parents.

File: docs/archive-redundancy.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+1. Upload block data to google cloud storage: To indicate a daemon to upload block data to google cloud storage, pass the flag `--upload-blocks-to-gcloud` . To successfully upload the file, daemon requires the following environment variables to be set:
+    1. `GCLOUD_KEYFILE` : Key file for authentication
+    2. `NETWORK_NAME`: Network name to be used in the filename
+    3. `GCLOUD_BLOCK_UPLOAD_BUCKET` : Google cloud storage bucket where the files are uploaded
+
+    The daemon generates a file for each block with the name `<network-name>-<protocol-state-hash>.json` . These are called precomputed blocks and will have all the fields of a block.
+
+2. Save block data from logs: The daemon also logs block data if the flag `-log-precomputed-blocks` is passed. The log to look for is `Saw block with state hash $state_hash` that contains `precomputed_block` in the metadata and has the block information. This is the same information (precomputed blocks) that gets uploaded to google cloud storage.
+3. Generate block data from another archive database: From a fully synced archive database, one can generate block data for each block using the `missing_subchain` tool.
```
Comment (psteckler) [inline docs/archive-redundancy.md:31]: nit: newline here?

File: docs/archive-redundancy.md (hunk 1 above)
Comment (psteckler) [reply docs/archive-redundancy.md:61]: the staged ledger is not a staking ledger, so maybe that should be described separately

also, the current syntax is just `staged-ledger`, not `current-staged-ledger`, and there's an optional `--state-hash` argument

**Review by bkase — APPROVED** [review]
once you address paul's comments it looks good -- but please put this on the website and not the code docs

**Comment by deepthiskumar** [comment]
Closing this. Added the docs to the website

**Pull request closed without merging**